### PR TITLE
luci-app-apinger: Add LuCI for Apinger

### PR DESCRIPTION
**Pull request by jempatel**
A new app for apinger with LuCI Support. This new LuCI app is added on top of the recently merged apinger PR
https://github.com/openwrt/packages/pull/18913, It covers backend UCI configuration information.

**We also need https://github.com/openwrt/packages/pull/19073**

Screenshots from the new Luci App for apinger:

Overview:
Apinger_Overview

Interfaces:
Apinger_Interfaces

Down Alarm:
Apinger_Alarm_Down

Delay Alarm:
Apinger_Alarm_Delay

Loss Alarm:
Apinger_Alarm_Loss

Targets:
Apinger_Targets

@feckert

### CODE REVIEW

File: applications/luci-app-apinger/htdocs/luci-static/resources/view/apinger/alarm_delay.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+'use strict';
+'require view';
+'require form';
+
+return view.extend({
+	load: function() {
+	},
+
+	renderDelayAlarms: function(m) {
+		var alarm_delay, alarm_delay_dl, alarm_delay_dh;
```
Comment (feckert) [inline applications/luci-app-apinger/htdocs/luci-static/resources/view/apinger/alarm_delay.js:10]: Please use the common `s m` pattern and not for every option and section an new variable name.

File: applications/luci-app-apinger/htdocs/luci-static/resources/view/apinger/targets.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		return m;
+	},
+
+	render: function(data) {
+		var m;
+
+		a_ifaces = uci.sections('apinger', 'interface');
+		a_down = uci.sections('apinger', 'alarm_down');
+		a_delay = uci.sections('apinger', 'alarm_delay');
+		a_loss = uci.sections('apinger', 'alarm_loss');
```
Comment (feckert) [inline applications/luci-app-apinger/htdocs/luci-static/resources/view/apinger/targets.js:72]: Why are these variables global can't we have them local?

File: applications/luci-app-apinger/htdocs/luci-static/resources/view/apinger/targets.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+		m = new form.Map('apinger', _('Apinger - Targets'),
+			_('Interface: Interface to use to track target') + '<br />' +
+			_('Address: Target address to be tracked') + '<br />' +
+			_('Ping Interval: How often the probe should be sent') + '<br />' +
+			_('Average Delay: How many replies should be used to compute average delay') + '<br />' +
+			_('Average Loss: How many probes should be used to compute average loss') + '<br />' +
+			_('Average Delay and Loss: The delay (in samples) after which loss is computed, without this delays larger than interval would be treated as loss') +
+			'<br />');
+		m = this.renderTargets(m);
```
Comment (feckert) [inline applications/luci-app-apinger/htdocs/luci-static/resources/view/apinger/targets.js:82]: Why do you have an extra function in all views? I would prefer the content of the called function to be inserted here. Instead of calling this. This would also allow you to keep the variables local.

**Review by feckert — CHANGES_REQUESTED** [review]

File: applications/luci-app-apinger/htdocs/luci-static/resources/view/apinger/alarm_delay.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+'use strict';
+'require view';
+'require form';
+
+return view.extend({
+	load: function() {
```
[the remark this answers is not in the document]
Comment (jempatel) [reply applications/luci-app-apinger/htdocs/luci-static/resources/view/apinger/alarm_delay.js]: Updated

File: applications/luci-app-apinger/htdocs/luci-static/resources/view/apinger/alarm_delay.js (hunk 1 above)
Comment (jempatel) [reply applications/luci-app-apinger/htdocs/luci-static/resources/view/apinger/alarm_delay.js]: Updated

File: applications/luci-app-apinger/htdocs/luci-static/resources/view/apinger/targets.js (hunk 2 above)
Comment (jempatel) [reply applications/luci-app-apinger/htdocs/luci-static/resources/view/apinger/targets.js]: Updated

File: applications/luci-app-apinger/htdocs/luci-static/resources/view/apinger/targets.js (hunk 3 above)
Comment (jempatel) [reply applications/luci-app-apinger/htdocs/luci-static/resources/view/apinger/targets.js]: Updated

File: applications/luci-app-apinger/root/usr/share/rpcd/acl.d/luci-app-apinger.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{
+	"luci-app-apinger" : {
+		"description" : "Grant access to LuCI app Apinger",
+		"read" : {
+			"ubus" : {
+				"apinger" : [ "*" ]
```
Comment (feckert) [inline applications/luci-app-apinger/root/usr/share/rpcd/acl.d/luci-app-apinger.json:6]: I think we also need a `read` and `write` uci acl for arpinger uci file

**Review by feckert — CHANGES_REQUESTED** [review]

File: applications/luci-app-apinger/root/usr/share/rpcd/acl.d/luci-app-apinger.json (hunk 5 above)
Comment (jempatel) [reply applications/luci-app-apinger/root/usr/share/rpcd/acl.d/luci-app-apinger.json:6]: Is it really needed? From LuCI, it's not reading/writing to /etc/config/apinger explicitly, instead, Its using RPC to get apinger overview.

Comment (feckert) [reply applications/luci-app-apinger/root/usr/share/rpcd/acl.d/luci-app-apinger.json:6]: As I see it, you are using the [uci handling](https://github.com/openwrt/luci/pull/5887/files#diff-43eb48c6a064931b875ac7df92656e5fef64b0d46101ec2d94b68b7984728768R4).
You can create a user and give him only the authorisation for the luci-app-apinger, then we will see if it also works without the uci acl.

Comment (jempatel) [reply applications/luci-app-apinger/root/usr/share/rpcd/acl.d/luci-app-apinger.json:6]: Yes, You are right. Updated rpcd ACL to handle apinger config.

File: applications/luci-app-apinger/Makefile [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+#
+# This is free software, licensed under the Apache License, Version 2.0 .
+#
+
+include $(TOPDIR)/rules.mk
+
+LUCI_TITLE:=LuCI support for the Apinger
+LUCI_DEPENDS:=+apinger
+LUCI_PKGARCH:=all
+PKG_LICENSE:=GPLv2
```
Comment (feckert) [inline applications/luci-app-apinger/Makefile:12]: Sorry, but can you use a valid `[SPDX](https://spdx.org/licenses/)` here?
maybe `PKG_LICENSE:=GPL-2.0` because `PKD_LICENSE:=GPLv2` is not valid

**Review by feckert — CHANGES_REQUESTED** [review]
Except for the license, it looks good :+1:

**Comment by jempatel** [comment]
Depends on PR
https://github.com/openwrt/packages/pull/19073

File: applications/luci-app-apinger/Makefile (hunk 6 above)
Comment (jempatel) [reply applications/luci-app-apinger/Makefile]: Updated.

**Comment by feckert** [comment]
Lets try it. Thanks for your pullrequest
